scripts/itest_parallel.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Get all the variables.
+PROCESSES=$1
+TRANCHES=$2
+TEST_FLAGS=$3
+ITEST_FLAGS=$4
+
+# Create a variable to hold the final exit code.
+exit_code=0
+
+# Run commands using xargs in parallel and capture their PIDs
+pids=()
+for ((i=0; i<PROCESSES; i++)); do 
+    cmd="scripts/itest_part.sh $i $TRANCHES $TEST_FLAGS $ITEST_FLAGS"
+    echo "$i" | xargs -r -P $PROCESSES -n 1 -I $i bash -c "$cmd " & pids+=($!)
+done
+
+
+# Wait for the processes created by xargs to finish.
+for pid in "${pids[@]}"; do
+    wait $pid
```

Comment:
I think we should use `wait -f` to make sure we wait for the process to _exit_ rather than just for the next change in status.

Reply:
I don't think `-f` works in all OS? For mac it says invalid option.

Reply:
Ah, the pleasure of writing portable shell scripts. Very possible that the BSD variant doesn't have that feature, yes.